morf-sqlserver/src/main/java/org/alfasoftware/morf/jdbc/sqlserver/SqlServerDialect.java

```diff
@@ -479,7 +506,28 @@ protected String getSqlFrom(UpdateStatement statement) {
     return sqlBuilder.toString();
   }
 
+
   /**
+   * @see SqlServerDialect#getSqlFrom(SequenceReference)
+   */
+  @Override
+    protected String getSqlFrom(SequenceReference sequenceReference) {
+    StringBuilder result = new StringBuilder();
+
+    switch (sequenceReference.getTypeOfOperation()) {
+      case NEXT_VALUE:
+        result.append("NEXT VALUE FOR ");
+        break;
+      case CURRENT_VALUE:
+        return "current_value FROM sys.sequences WHERE name = '" + sequenceReference.getName() + "'";
+    }
+
+    result.append(sequenceReference.getName());

```

Comment:
This does not work with CURRENT_VALUE, only NEXT_VALUE should use it.
Move this directly to the NEXT_VALUE case.
Check what the test will spit out when used.

Reply:
Oh, I see the return in CURRENT_VALUE instead of a break!
That's okay then, but you got me! :)